scipy/stats/_wilcoxon.py

```diff
@@ -111,15 +111,6 @@ def _wilcoxon_iv(x, y, zero_method, correction, alternative, method, axis):
     if method == "auto" and d.shape[-1] > 50:
         method = "asymptotic"
 
-    if n_zero > 0 and method == "exact":
-        warnings.warn("Exact p-value calculation does not work if there are "
-                      "zeros. Consider using method `asymptotic` or "
-                      "`stats.PermutationMethod`",
-                      stacklevel=2)
```

Comment:
As noted in https://github.com/scipy/scipy/pull/21567#discussion_r1810899084:

- `'exact'` is also not exact if there are ties, but there is no similar warning in that case. 
-  Users have expressed confusion about what a "zero" is in the past. 
- The recommended normal approximation / `method='asymptotic'` is not exact, either, so this doesn't motivate why one would be better than the other. (A permuation test would also not be exact unless the sample size is small.)
 
It's tough to address all the subtleties of accurate hypothesis testing in a warning; that's why we've elaborated on this stuff in the documentation. And I'm concerned that incomplete warnings like this can do more harm than good: it may give users the impression that the code will help them avoid shooting themselves in the foot, but `scipy.stats` doesn't do that consistently. If we think a more consistent policy across the board is important, that is something we can consider, but for now, let's not give a false impression.